src/BuildScriptGeneratorCli/Commands/BuildCommand.cs

```diff
@@ -79,18 +90,89 @@ public string LanguageVersion
+            var packageOption = new Option<bool>(OptionTemplates.Package, OptionTemplates.PackageDescription);
+            var osReqOption = new Option<string>(OptionTemplates.OsRequirements, OptionTemplates.OsRequirementsDescription);
+            var appTypeOption = new Option<string>(OptionTemplates.AppType, OptionTemplates.AppTypeDescription);
+            var buildCommandFileNameOption = new Option<string>(OptionTemplates.BuildCommandsFileName, OptionTemplates.BuildCommandsFileNameDescription);
+            var compressDestDirOption = new Option<bool>(OptionTemplates.CompressDestinationDir, OptionTemplates.CompressDestinationDirDescription);
+            var propertyOption = new Option<string[]>(OptionTemplates.Property, OptionTemplates.PropertyDescription);
+            var dynamicInstallRootDirOption = new Option<string>(OptionTemplates.DynamicInstallRootDir, OptionTemplates.DynamicInstallRootDirDescription);
+
+            // Hiding Language Option because it is obselete
+            var languageOption = new Option<string>(OptionTemplates.Language, OptionTemplates.LanguageDescription);
+            languageOption.IsHidden = true;
+
+            // LanguageVer Option is obselete
+            var languageVerOption = new Option<string>(OptionTemplates.LanguageVersion, OptionTemplates.LanguageVersionDescription);
+            languageVerOption.IsHidden = true;
+            var intermediateDirOption = new Option<string>(OptionTemplates.IntermediateDir, OptionTemplates.IntermediateDirDescription);
+            var outputOption = new Option<string>(aliases: OptionTemplates.Output, OptionTemplates.OutputDescription);
+            var manifestDirOption = new Option<string>(OptionTemplates.ManifestDir, OptionTemplates.ManifestDirDescription);
+
+            var command = new Command("build", "Build an app.")
+            {
+                sourceDirArgument,
+                intermediateDirOption,
+                outputOption,
+                manifestDirOption,
+                platformOption,
+                platformVersionOption,
+                packageOption,
+                osReqOption,
+                appTypeOption,
+                buildCommandFileNameOption,
+                compressDestDirOption,
+                propertyOption,
+                dynamicInstallRootDirOption,
+                languageOption,
+                languageVerOption,
+                logOption,
+                debugOption,
+            };
+
+            command.SetHandler(
+                (prop) =>
+                {
+                    // InvocationContext provided in SetHandler
+                    var buildCommand = new BuildCommand(prop);
+                    var returnCode = buildCommand.OnExecute(console);
+                    return Task.FromResult(returnCode);
+                },
+                new BuildCommandBinder(
+                    languageOption,
```

Comment:
One suggestion that would be nice and would help to prevent future incidents: when you're initializing an object and providing a large amount of arguments to the constructor (or even with a method that has a lot of arguments), it's best to specify the argument name before each value you provide so it's clear which argument each value is being mapped to.

This is especially useful if we were to introduce a new argument to this constructor in the future and it wasn't necessarily tacked on to the end, there's a possibility the value gets mapped to the wrong argument.

I would do this for each of the `Binder` objects that you're initializing for the `SetHandler` call in each command class.